ecosystem/typescript/sdk/src/plugins/fungible_asset_client.ts

```diff
@@ -0,0 +1,113 @@
+import { AptosAccount } from "../account";
+import { RawTransaction } from "../aptos_types";
+import * as Gen from "../generated/index";
+import { OptionalTransactionArgs, Provider } from "../providers";
+import { TransactionBuilderRemoteABI } from "../transaction_builder";
+import { MaybeHexString, HexString } from "../utils";
+
+export class FungibleAssetClient {
+  provider: Provider;
+
+  readonly assetType: string = "0x1::fungible_asset::Metadata";
+
+  /**
+   * Creates new FungibleAssetClient instance
+   *
+   * @param provider Provider instance
+   */
+  constructor(provider: Provider) {
+    this.provider = provider;
+  }
+
+  /**
+   * Transfer an asset amount from the sender primary_fungible_store to the recipient primary_fungible_store
+   *
+   * Use this method to transfer any fungible asset including fungible token.
+   *
+   * @param sender The sender account
+   * @param assetAddress The fungible asset address.
+   * For example if you’re transferring USDT this would be the USDT address
+   * @param recipient Recipient address
+   * @param amount Number of assets to transfer
+   * @param assetType (optional) The fungible asset type - default to `0x1::fungible_asset::Metadata`
+   * @returns The hash of the transaction submitted to the API
+   */
+  async transferAmount(
+    sender: AptosAccount,
+    assetAddress: MaybeHexString,
+    recipient: MaybeHexString,
+    amount: number | bigint,
+    assetType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    const rawTxn = await this.generateTransferAmount(sender, assetAddress, recipient, amount, assetType, extraArgs);
+    const txnHash = await this.submit(sender, rawTxn);
+    return txnHash;
+  }
+
+  /**
+   * Get the balance of an account's fungible asset.
```

Comment:
We should document that this looks at an account's primary store, which would exclude their balances in all other untracked stores. cc @lightmark. We should provide a separate function (something like storeBalance) that takes a store address and return the balance.

Reply:
Yes, I refactored managed_fungible_asset example. @0xmaayan We can talk about this on slack.

Reply:
now that we have `check balance for primary store` and `check balance for secondary store` I guess having the `coin_client.balance()` function support both or one of them doesn't make sense.